app/models/user.rb

```diff
@@ -0,0 +1,85 @@
+class User < ApplicationRecord
+  attr_accessor :remember_token, :activation_token, :reset_token
+  before_save   :downcase_email
+  before_create :create_activation_digest
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
+  has_many :orders
+
+class << self
+  # Returns the hash digest of the given string.
+  def digest(string)
```

Comment:
Aren't we using devise?

Reply:
I am not using devise, I did the login system from Rails tutorial by Michael Hartl 😁